### PR TITLE
Added EntityTotemActivateEvent

### PR DESCRIPTION
**Pull request by VendoAU**
The mixin is a bit ugly, but I don't think there's a better way to do it.

### CODE REVIEW

**Comment by VendoAU** [comment]
I have no idea why I thought it was better to inject inside the not null check.
I've made all the changes you've asked for.

File: src/main/java/xyz/nucleoid/stimuli/mixin/entity/LivingEntityMixin.java [hunk 1]
```diff
@@ -82,4 +83,15 @@ private void modifyDroppedLoot(LootTable instance, LootContextParameterSet param
             }
         }
     }
+
+    @Inject(method = "tryUseTotem", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;decrement(I)V"), cancellable = true)
+    private void tryUseTotem(DamageSource source, CallbackInfoReturnable<Boolean> cir, @Local(ordinal = 1) ItemStack itemStack) {
```
[the remark this answers is not in the document]
Comment (VendoAU) [reply src/main/java/xyz/nucleoid/stimuli/mixin/entity/LivingEntityMixin.java:88]: The `this.getStackInHand(hand)` item stack is the one that is captured, which is the same one that is decremented.

Here's a snippet I got using `@Debug(export = true)`:
```java
for(int var6 = 0; var6 < var5; ++var6) {
    Hand hand = var4[var6];
    ItemStack itemStack2 = this.getStackInHand(hand);
    if (itemStack2.isOf(Items.TOTEM_OF_UNDYING)) {
        itemStack = itemStack2.copy();
        CallbackInfoReturnable callbackInfo8 = new CallbackInfoReturnable("tryUseTotem", true);
        this.handler$zgl000$stimuli$tryUseTotem(source, callbackInfo8, itemStack2);
        if (callbackInfo8.isCancelled()) {
            return callbackInfo8.getReturnValueZ();
        }
        itemStack2.decrement(1);
        break;
    }
}
```
I tested and the totem only gets decremented on the second case which is correct.

[the remark this answers is not in the document]
Comment (VendoAU) [reply src/main/java/xyz/nucleoid/stimuli/mixin/entity/LivingEntityMixin.java:88]: I've added this now

**Comment by MushroomMif** [comment]
Hi! I was looking at code of this PR and noticed that at the start of new `tryUseTotem` method it returns if the world is not a client world:
```
if (!this.getWorld().isClient) {
    return;
}
```
I think this is unexpected and it should return if the world **is** a client world, isn't it?

**Comment by VendoAU** [comment]
> Hi! I was looking at code of this PR and noticed that at the start of new `tryUseTotem` method it returns if the world is not a client world:
> 
> ```
> if (!this.getWorld().isClient) {
>     return;
> }
> ```
> 
> I think this is unexpected and it should return if the world **is** a client world, isn't it?

You're right, I've fixed this now

**Review by Gegy — APPROVED** [review]

**Pull request merged**